server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -423,11 +431,35 @@ protected void setMinSeqNoToKeep(long seqNo) {
         this.minSeqNoToKeep = seqNo;
     }
 
+    @Override
+    protected Releasable drainSync() {
+        try {
+            if (syncPermits.tryAcquire(SYNC_PERMITS, 1, TimeUnit.MINUTES)) {
+                logger.info("All permits acquired");
+                return Releasables.releaseOnce(() -> {
+                    syncPermits.release(SYNC_PERMITS);
+                    assert syncPermits.availablePermits() == SYNC_PERMITS : "Available permits is " + syncPermits.availablePermits();
+                    logger.info("All permits released");
+                });
+            } else {
+                throw new TimeoutException("Timeout while acquiring all permits");
+            }
+        } catch (TimeoutException | InterruptedException e) {
+            throw new RuntimeException("Failed to acquire all permits", e);
+        }
```

Comment:
1. Move logger to debug or provide more context
2. Does 1 min align with SDK retries and timeouts, I doubt?

Reply:
For no 1, adding more context in the logs. For no 2, when this method is called during primary relocation, the translog upload has already finished for all indexing that has happened until the indexing block kicked in. This acquire is to ensure that if there is any ongoing internal flush (triggered by flushOnIdle, Snapshot or similar) which has no additional data, should finish before we start the force segrep step on the new primary. This has been set a bit lenient for general happy cases across all possible repository plugin implementation.